src/utils.ts

```diff
@@ -74,17 +75,59 @@ export function isW3CThingDescription(thingDescription: {
 }): boolean {
   if (typeof thingDescription['@context'] === 'string') {
     return (
-      thingDescription['@context'] === 'https://www.w3.org/2019/wot/td/v1' ||
-      thingDescription['@context'] === 'http://www.w3.org/ns/td'
+      thingDescription['@context'] === Constants.WOT_TD_1_CONTEXT ||
+      thingDescription['@context'] === Constants.WOT_TD_1_1_CONTEXT ||
+      thingDescription['@context'] === Constants.WOT_TD_NS_CONTEXT
     );
   }
 
   if (Array.isArray(thingDescription['@context'])) {
     return (
-      thingDescription['@context'].includes('https://www.w3.org/2019/wot/td/v1') ||
-      thingDescription['@context'].includes('http://www.w3.org/ns/td')
+      thingDescription['@context'].includes(Constants.WOT_TD_1_CONTEXT) ||
+      thingDescription['@context'].includes(Constants.WOT_TD_1_1_CONTEXT) ||
+      thingDescription['@context'].includes(Constants.WOT_TD_NS_CONTEXT)
     );
   }
 
   return false;
 }
+
+/**
+ * Make sure the @context value of a Thing Description contains both
+ * W3C WoT TD 1.1 and WebThings contexts.
+ *
+ * @param context {(string|string[])} The non-standardised context value.
+ * @returns string[] The standardised context array or null if already OK.
+ */
+export function standardizeContext(context: string | string[]): string[] | null {
+  // If expected contexts already present then return null
+  if (
+    Array.isArray(context) &&
+    context.includes(Constants.WOT_TD_1_1_CONTEXT) &&
+    context.includes(Constants.WEBTHINGS_CONTEXT)
+  ) {
+    return null;
```

Comment:
Why return `null`? can't we just return the context as it is? it should simplify https://github.com/WebThingsIO/gateway/pull/3034/files#diff-298926bb41b428e204128605377a50dec34c4e90cc601ac06bd172563a91a399R717 . Given the fact that the contract of the function is to produce a standardized context, if it is already standard it makes sense to return it without touching.

Reply:
Good point! Fixed.